torch/csrc/dynamo/guards.cpp

```diff
@@ -3405,6 +3405,29 @@ class TupleIteratorGetItemAccessor : public GuardAccessor {
   Py_ssize_t _index;
 };
 
+// Gets weakref ref (strong ref) on construction,
+// releases the ref on destruction.
+class WeakrefRAII {
+  PyObject* _obj{nullptr};
+
+ public:
+  WeakrefRAII(PyObject* weakref) {
+    PyWeakref_GetRef(weakref, &_obj); // strong ref
```

Comment:
You need to check the return value here, see:
https://docs.python.org/3/c-api/weakref.html#c.PyWeakref_GetRef

Reply:
Ah I assumed that `_obj` would remain `nullptr` on error and an exception would be set in Python, which we would propagate upward.